yarn-project/aztec-nr/address-note/src/address_note.nr

```diff
@@ -8,33 +8,51 @@ use dep::aztec::{
         utils::compute_siloed_note_hash,
     },
     oracle::{
+        rand::rand,
         get_secret_key::get_secret_key,
         get_public_key::get_public_key,
     },
     hash::pedersen_hash,
     context::PrivateContext,
 };
 
-global ADDRESS_NOTE_LEN: Field = 2;
+global ADDRESS_NOTE_LEN: Field = 3;
 
 // Stores an address
 struct AddressNote {
     address: Field,
     owner: Field,
+    randomness: Field,
     header: NoteHeader,
 }
 
 impl AddressNote {
     pub fn new(address: Field, owner: Field) -> Self {
+        let randomness = rand();
         AddressNote {
             address,
             owner,
+            randomness,
             header: NoteHeader::empty(),
         }
     }
 
     pub fn serialize(self) -> [Field; ADDRESS_NOTE_LEN]{
-        [self.address, self.owner]
+        [self.address, self.owner, self.randomness]
+    }
+
+    pub fn deserialize(preimage: [Field; ADDRESS_NOTE_LEN]) -> Self {
+        AddressNote {
+            address: preimage[0],
+            owner: preimage[1],
+            randomness: preimage[2],
+            header: NoteHeader::empty(),
+        }
+    }
+
+    pub fn compute_note_hash(self) -> Field {
```

Comment:
wouldn't the impl of compute_note_hash and nullifier be slightly different such that you can still get the note but not the nullifer?

Here looks like you would need to know the randomness to get the note?

Reply:
Yes, you need to know the randomness to get the note hash but I think this is quite normal. The difference between getting a nullifier and just a note hash is that for nullifier you need to know the owner secret.

BTW after [this commit](https://github.com/AztecProtocol/aztec-packages/pull/2965/commits/c634cf095d378cfd17c8d56f0fc6ece014e4aef6) the note is basically just a copy of ValueNote so I think all should be good here.